### PR TITLE
Constrain packages for Lwt 4.0.0

### PR DESCRIPTION
**Pull request by aantron**
See https://github.com/ocaml/opam-repository/pull/11708#issuecomment-377951313.

cc @hannesm 

I'll rebase the "main" Lwt 4.0.0 PR over this, once this is merged.

### CODE REVIEW

**Comment by hannesm** [comment]
I think this is fine.  Would really like to get an opinion from @camelus in here.

the CI failures are -- as far as i could see -- not related to the lwt constraints, but rather due to other breakages

unless someone objects within the next 24 hours, i'll merge this PR to level the path for lwt 4.0.0 in the main opam repository

File: packages/jupyter-kernel/jupyter-kernel.0.1/opam [hunk 1]
```diff
@@ -16,7 +16,7 @@ depends: [
   "atdgen"
   "yojson"
   "uuidm"
-  "lwt"
+  "lwt" {< "4.0.0"]
```
Comment (avsm) [inline packages/jupyter-kernel/jupyter-kernel.0.1/opam]: there is a typo here that was killing the opam2 test (missing closing brace)

**Comment by avsm** [comment]
this lgtm once the opam2 conversion passes successfully in CI (not necessarily all the builds themselves)

**Comment by aantron** [comment]
Oops, fixed. Thanks @avsm.

This PR should probably be squash-merged, then.

File: packages/qcow/qcow.0.6.0/opam [hunk 2]
```diff
@@ -15,7 +15,7 @@ depends: [
   "cstruct"
   "result"
   "mirage-types-lwt" {>="2.6.0" & < "3.0.0"}
-  "lwt"
+  "lwt" {< "4.0.0"}
```
Comment (avsm) [inline packages/qcow/qcow.0.6.0/opam]: also missing brace here

File: packages/qcow/qcow.0.7.0/opam [hunk 3]
```diff
@@ -15,7 +15,7 @@ depends: [
   "cstruct"
   "result"
   "mirage-types-lwt" {>="2.6.0" & < "3.0.0"}
-  "lwt"
+  "lwt" {< "4.0.0"]
```
Comment (avsm) [inline packages/qcow/qcow.0.7.0/opam]: and here

File: packages/qcow/qcow.0.6.0/opam (hunk 2 above)
Comment (aantron) [reply packages/qcow/qcow.0.6.0/opam]: Don't see it in this one

**Comment by aantron** [comment]
Argh, I was hoping for @camelus. I don't mean for you to be a build bot @avsm, let me look through the remaining constraints.

**Comment by avsm** [comment]
That's all right -- the bot won't spot some of these typos -- the opam2 conversion script is far stricter however, so making sure that roughly the same packages are attempted to be installed with both the opam1 and 2 CI paths is useful for these bigger patches.

**Comment by aantron** [comment]
Ok.

I used `opam lint` and found no more errors, but it did spot both of the ones above (with File does not parse).

In case you need to recommend someone to do this in the future, the command was:

```
git diff HEAD~ HEAD --name-only | xargs -L1 opam lint
```

**Comment by hannesm** [comment]
thanks @aantron, merging... please rebase your lwt 4.0 PR on master to check the revdeps builds

**Pull request merged**